src/bin/pg_autoctl/pgctl.c

```diff
@@ -918,6 +1030,100 @@ escape_recovery_conf_string(char *destination, const char *recoveryConfString)
 }
 
 
+/*
+ * prepare_primary_conninfo
+ */
+static int
+prepare_primary_conninfo(char *primaryConnInfo, int primaryConnInfoSize,
+						 const char *primaryHost, int primaryPort,
+						 const char *replicationUsername,
+						 const char *replicationPassword)
+{
+	char primaryConnInfoRaw[MAXCONNINFO] = { 0 };
+	char *connInfoEnd = primaryConnInfoRaw;
+
+ 	connInfoEnd += make_conninfo_field_str(connInfoEnd, "host", primaryHost);
```

Comment:
please consider using StringInfo if possible. using hardcoded lengths and unchecked accesses always worry me.